### ISSUE TITLE
Make dragonfly "Text" action work with nordic keyboards

### ISSUE THREAD

**Issue by madsBR** [t4ngo/dragonfly#67]
Atm the Text action bugs with all chars which require alt+ctrl on the Keyboard. On a nordic keyboard this includes 
@£$€{[]}|

i fixed it by replacing 
"elif" by "if" in here
https://github.com/t4ngo/dragonfly/blob/master/dragonfly/actions/keyboard.py
at line 121,122,137,138

but i don't know if it will have bug -consequences later..

**Comment by Shoggomo** [comment]
None of these forks worked for me (german keyboard). Monospark's fork doesn't even seem to work for my usecase (I'm using caster ontop of dragonfly). However 8327d756d15f3d7235fca5404e73ebc3e6efda2e from Monospark's fork seems to be the language fix and could be implemented in another fork.

**Comment by chilimangoes** [comment]
Yep, that was the change set I was referring to. @pimp22 have you tried the above solution to see if it works for your issues?

**Comment by reckoner** [comment]
Does this apply to the case where,

    Text('σμ').execute()

would work? That is, does it get `Text` to emit Unicode UTF-8 characters?

**Comment by Danesprite** [comment]
@reckoner No, unfortunately this doesn't work for all UTF-8 characters. `Text` and `Key` are dependent on the keyboard layout you use. The changes mentioned just allow some different layouts to work correctly.

**Comment by reckoner** [comment]
@Danesprite  Thanks! You just saved me a lot of research.

**Comment by Danesprite** [comment]
@reckoner No problem. I think the Windows API does allow for that, but it requires rewriting dragonfly's keyboard interface. One workaround for this is to use the `Clipboard` class to copy the Unicode strings you want and paste them using  `Key`. E.g.:

```Python
from dragonfly import Clipboard, Key
Clipboard.set_system_text(u"σμ")
Key("c-v").execute()

```

I think you can also use the `Paste` action.

You need to put `# encoding=utf-8` (or similar) on the first line of the source file to make Python 2.7 use that encoding instead.

**Comment by Versatilus** [comment]
> @reckoner No problem. I think the Windows API does allow for that, but it requires rewriting dragonfly's keyboard interface.

I have a mostly functional implementation of this I've been using for most of 2018. It didn't require rewriting nearly as much as I expected it to. I'm not sure that I've even bothered to push it to GitHub. I'm up for discussing the issues in Gitter when I've finally finished this SimCity binge I've been on.

**Comment by Danesprite** [comment]
Ah okay then, that's good to hear. Sounds good :+1:

**Comment by reckoner** [comment]
@Danesprite Thanks for the `Paste` idea. I will look into it. I also look forward to your @Versatilus  keyboard implementation.

**Comment by caspark** [comment]
For the record, it looks like @Versatilus has merged [his Unicode Text action PR](https://github.com/Danesprite/dragonfly/pull/44) into @Danesprite 's [fork](https://github.com/Danesprite/dragonfly), so uninstalling original dragonfly then doing something like `pip install dragonfly2==0.10 --upgrade` might do the trick for you now @reckoner .
